cmd/gcp-controller-manager/dpwi/serviceaccounts/verifier.go

```diff
@@ -0,0 +1,198 @@
+package serviceaccounts
+
+import (
+	"fmt"
+	"sync"
+
+	"golang.org/x/sync/singleflight"
+	core "k8s.io/api/core/v1"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/cloud-provider-gcp/cmd/gcp-controller-manager/dpwi/hms"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// ServiceAccountAnnotationGSAEmail is the key to GCP Service Account annotation in
+	// ServiceAccount objects.
+	ServiceAccountAnnotationGSAEmail = "iam.gke.io/gcp-service-account"
+)
+
+// ServiceAccountVerifier implements a custom control loop responsible for verifying authorization
```

Comment:
This isn't a control loop, right?

Reply:
You are right. The comment was copied and I didn't update it. Updated.